File: languages/python/bloock/entity/integrity/anchor.py
from __future__ import annotations

from typing import List

import bloock._bridge.proto.integrity_entities_pb2 as proto
from bloock.entity.integrity.anchor_network import AnchorNetwork


class Anchor:
    """
    Represents information about an anchor.
    """
    def __init__(
            self,
            id: int,
            block_roots: List[str],
            networks: List[AnchorNetwork],
            root: str,
            status: str,
    ) -> None:
        """
        Constructs an Anchor object with the specified parameters.
        :type status: object
        :type root: object
        :type networks: object
        :type block_roots: object
        :type id: object
        :rtype: object
        """
        self.id = id
        self.block_roots = block_roots
        self.networks = networks
        self.root = root
        self.status = status

    @staticmethod
    def from_proto(anchor: proto.Anchor) -> Anchor:
        return Anchor(
            id=anchor.id,
            block_roots=list(anchor.block_roots),
            networks=list(map(lambda x: AnchorNetwork.from_proto(x), anchor.networks)),
            root=anchor.root,
            status=anchor.status,
        )

    def to_proto(self) -> proto.Anchor:
        return proto.Anchor(
            id=self.id,
            block_roots=list(self.block_roots),
            networks=list(map(lambda x: x.to_proto(), self.networks)),
            root=self.root,
            status=self.status,
        )

File: languages/php/Bloock/Entity/Integrity/Anchor.php
<?php

namespace Bloock\Entity\Integrity;

use Google\Protobuf\Internal\RepeatedField;

/**
 * Represents information about an anchor.
 */
class Anchor
{
    private int $id;
    private RepeatedField $blockRoots;
    private RepeatedField $networks;
    private string $root;
    private string $status;

    /**
     * Constructs an Anchor object with the specified parameters.
     * @param int $id
     * @param RepeatedField $blockRoots
     * @param RepeatedField $networks
     * @param string $root
     * @param string $status
     */
    public function __construct(int $id, RepeatedField $blockRoots, RepeatedField $networks, string $root, string $status)
    {
        $this->id = $id;
        $this->blockRoots = $blockRoots;
        $this->networks = $networks;
        $this->root = $root;
        $this->status = $status;
    }

    public static function fromProto(\Bloock\Anchor $anchor): Anchor
    {
        return new Anchor($anchor->getId(), $anchor->getBlockRoots(), $anchor->getNetworks(), $anchor->getRoot(), $anchor->getStatus());
    }

    /**
     * Gets id from the anchor.
     * @return int
     */
    public function getId(): int
    {
        return $this->id;
    }

    /**
     * Gets block roots from the anchor.
     * @return RepeatedField
     */
    public function getBlockRoots(): RepeatedField
    {
        return $this->blockRoots;
    }

    /**
     * Gets networks from the anchor.
     * @return RepeatedField
     */
    public function getNetworks(): RepeatedField
    {
        return $this->networks;
    }

    /**
     * Gets root from the anchor.
     * @return string
     */
    public function getRoot(): string
    {
        return $this->root;
    }

    /**
     * Get status from the anchor.
     * @return string
     */
    public function getStatus(): string
    {
        return $this->status;
    }

    public function toProto(): \Bloock\Anchor
    {
        $p = new \Bloock\Anchor();
        $p->setId($this->id);
        $p->setBlockRoots($this->blockRoots);

        $n = [];
        foreach ($this->networks->getIterator() as $network) {
            $n[] = $network->toProto();
        }
        $p->setNetworks($n);
        return $p;
    }
}